### PR TITLE
Python SocketIO Example 

### PR DESCRIPTION
**Pull request by colton-smith**
### Socketio
This PR adds a simple example for python socketio that is a pared down version of the telemetry and logging socketio clients used in the PDS circa competition 4.

See the readme for a description of each file, how to run the example, and some notes about socketio.

### CODE REVIEW

**Comment by Burke-Daniel** [comment]
@colton-smith you still planning on making this use flask socketio? If so I'll wait to review

File: Examples/socketio-python/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+`client_logger.py`
+- Socketio client that listens for event "log_telemetry" and logs
+the data packet to console
+
+`client_telemetry.py`
+- Socketio client that generates telem (simulates receiving from the TBM) and emits a "log_telemetry" event containing the data packet
+
+`server.py`
+- Standard socketio server, handles event routing between clients
```
Comment (Burke-Daniel) [inline Examples/socketio-python/README.md:15]: Does not really matter that much, but should we mention now that this is flask-socketio?

File: Examples/socketio-python/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@sio.event
+def my_event(sid, data):
+    pass
+
+# OR
+
+@sio.on('my event name')
+def another_event(sid, data):
+    pass
+```
```
Comment (Burke-Daniel) [inline Examples/socketio-python/README.md:40]: In flask-socketio, you have to use `@sio.on()` though right?

File: Examples/socketio-python/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+For asyncio servers, event handlers can optionally be given as coroutines:
+
+```
+@sio.event
+async def my_event(sid, data):
+    pass
+```
+
+The sid arguement is the socketio session id, a unique identifier of each client connection. All events sent by a given client will have the same sid value.
```
Comment (Burke-Daniel) [inline Examples/socketio-python/README.md:53]: *argument

File: Examples/socketio-python/client_logger.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@sio.event
+def disconnect():
+    print("Disconnected from server")
+    
+    
+@sio.on("telemetry")
+def telemetry(data):
+    print(f"TELEM RECEIVED - {data}")
```
Comment (Burke-Daniel) [inline Examples/socketio-python/client_logger.py:28]: This question is just out of curiosity, but could this also be:
```python
@sio.event
def telemetry(data):
```

File: Examples/socketio-python/server.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+from flask import Flask, render_template
+from flask_socketio import SocketIO
+    
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'secret!'
```
Comment (Burke-Daniel) [inline Examples/socketio-python/server.py:5]: lolol very secret indeed

**Review by Burke-Daniel — APPROVED** [review]
Just a couple little questions/comments, but either way this looks GTM now whenever!

File: Examples/socketio-python/README.md (hunk 1 above)
Comment (colton-smith) [reply Examples/socketio-python/README.md:15]: True - resolved

File: Examples/socketio-python/README.md (hunk 2 above)
Comment (colton-smith) [reply Examples/socketio-python/README.md:40]: True - resolved

File: Examples/socketio-python/client_logger.py (hunk 4 above)
Comment (colton-smith) [reply Examples/socketio-python/client_logger.py:28]: Yeah the function name doesnt matter when using on, only with event

**Pull request merged**